cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+            host,
+            principal,
+            clusterAclRequest.getRequestOperationType().value);
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+      return ApiResultStatus.FAILURE.value;
+    }
+
+    return ApiResultStatus.SUCCESS.value;
+  }
+
+  private void processOtherRequests(
+      ClusterAclRequest clusterAclRequest,
+      AdminClient client,
+      PatternType patternType,
+      String host,
+      String principal)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    List<AclBindingFilter> aclListArray = new ArrayList<>();
+
+    ResourcePatternFilter resourcePattern =
+        new ResourcePatternFilter(
+            ResourceType.TOPIC, clusterAclRequest.getTopicName(), patternType);
+    AccessControlEntryFilter aclEntry =
+        new AccessControlEntryFilter(principal, host, AclOperation.WRITE, AclPermissionType.ALLOW);
+    AclBindingFilter aclBinding1 = new AclBindingFilter(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding1);
+
+    aclEntry =
+        new AccessControlEntryFilter(
+            principal, host, AclOperation.DESCRIBE, AclPermissionType.ALLOW);
+    AclBindingFilter aclBinding2 = new AclBindingFilter(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding2);
+
+    client.deleteAcls(aclListArray).all().get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS);
+  }
+
+  private boolean updateTopicProducerWriteAcls(
+      String topicName, AdminClient client, PatternType patternType, String host, String principal)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    List<AclBinding> aclListArray = new ArrayList<>();
+
+    ResourcePattern resourcePattern =
+        new ResourcePattern(ResourceType.TOPIC, topicName, patternType);
+    AccessControlEntry aclEntry =
+        new AccessControlEntry(principal, host, AclOperation.WRITE, AclPermissionType.ALLOW);
+    AclBinding aclBinding1 = new AclBinding(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding1);
+
+    boolean acl1Exists = aclExists(client, aclBinding1.toFilter());
+
+    aclEntry =
+        new AccessControlEntry(principal, host, AclOperation.DESCRIBE, AclPermissionType.ALLOW);
+    AclBinding aclBinding2 = new AclBinding(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding2);
+
+    boolean acl2Exists = aclExists(client, aclBinding2.toFilter());
+
+    if (acl1Exists && acl2Exists) {
+      return true;
+    } else {
+      client.createAcls(aclListArray).all().get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS);
+      return false;
+    }
+  }
+
+  private boolean aclExists(AdminClient client, AclBindingFilter aclBindingFilter) {
+    DescribeAclsResult aclsResult = client.describeAcls(aclBindingFilter);
+    try {
+      if (aclsResult.values().get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS).size() == 1) {
+        return true;
+      }
+    } catch (InterruptedException | TimeoutException | ExecutionException e) {
+      e.printStackTrace();
```

Comment:
```suggestion
     log.error("Exception:", e);
```
It's better to log into logger to be able to find it in future for possible analysis